File: LinkedList/1.js


// let head ={
//     data:10,
//     next:{
//         data:90,
//         next:{
//             data:43,
//             next:null
//         }

//     }
// }

// console.log(head)

class Node {
    constructor(data){
        this.data=data
        this.next=null
    }
}


class LinkedList {
    constructor(data){
        this.head=new Node(data)
        this.tail=this.head
        this.length=1;
    }

    append(data){
        let t=new Node(data)
        this.tail.next=t
        this.tail=t
        this.length++

    }
}


let n1=new LinkedList(100)
n1.append(200)

console.log(n1)



// let q=new Node(23)
// // console.log(q)
// let w=new Node(34)
// // console.log(w)
// q.next=w
// console.log(q)

File: LinkedList/1.cpp
#include<bits/stdc++.h>
using namespace std;
struct Node{
    int data ;
    struct Node *next;
}*first=NULL;

void create(int A[],int n){
    struct Node *last,*t;
    first=new Node;
    first->data=A[0];
    first->next=NULL;
    last=first;
    for(int i=1;i<n;i++){
        t=new Node; 
        t->data=A[i];
        t->next=NULL;
        last->next=t;
        last=t;
    }
}

void RDisplay(struct Node *p){
    if(p!=NULL){
        cout<<p->data<<' ';
        RDisplay(p->next);
    }
    // cout<<endl;
}

void display(struct Node *p){
    while(p!=NULL){
        cout<<p->data<<' ';
        p=p->next;
    }
    cout<<endl;
}

int main(){
    int A[]={4,5,1,9};
    create(A,4);
    display(first);

}